src/core/token/TokenManager.ttslua

```diff
@@ -0,0 +1,446 @@
+      TokenManager.spawnToken(offsets[i], tokenType, card.getRotation())
+    end
+  end
+
+  -- Spawns a single token at the given global position by copying it from the template bag.
+  ---@param position Global position to spawn the token
+  ---@param tokenType String type of token to spawn, valid values are "damage", "horror",
+  --     "resource", "doom", or "clue"
+  ---@param rotation Vector Rotation to be used for the new token.  Only the y-value will be used,
+  --     x and z will use the default rotation from the source bag
+  ---@param callback A callback function triggered after the new token is spawned
+  TokenManager.spawnToken = function(position, tokenType, rotation, callback)
+    if SOURCE_BAG_GUIDS[tokenType] == nil then
+      error("Unknown token type '" .. tokenType .. "'")
+      return
+    end
+    local sourceBag = getObjectFromGUID(SOURCE_BAG_GUIDS[tokenType])
+    if sourceBag == nil then
+      error("No token source for '" .. tokenType .. "'")
+      return
+    end
+    -- All the source bags are infinite, so we just grab the first object
+    local tokenTemplate = sourceBag.getData().ContainedObjects[1]
+    -- Take ONLY the Y-value for rotation, so we don't flip the token coming out of the bag
+    local tokenRotation  = rotation or { x = 0, y = 270, z = 0 }
+    tokenTemplate.Transform.rotY = tokenRotation.y
+    log("x=" .. tokenTemplate.Transform.rotX .. ",y=" .. tokenTemplate.Transform.rotY .. "z=" .. tokenTemplate.Transform.rotZ)
+    return spawnObjectData({
+      data = tokenTemplate,
+      position = position,
+      callback_function = callback
+    })
+  end
+
+  -- Delegate function to the token spawn tracker.  Exists to avoid circular dependencies in some
+  -- callers.
+  ---@param card Object Card object to reset the tokens for
+  TokenManager.resetTokensSpawned = function(card)
+    tokenSpawnTracker.resetTokensSpawned(card.getGUID())
+  end
+
+  -- Pushes new player card data into the local copy of the Data Helper player data.
+  ---@param dataTable Table Key/Value pairs following the DataHelper style
+  TokenManager.addPlayerCardData = function(dataTable)
+    internal.initDataHelperData()
+    for k, v in pairs(dataTable) do
+      playerCardData[k] = v
+    end
+  end
+
+  -- Pushes new location data into the local copy of the Data Helper location data.
+  ---@param dataTable Table Key/Value pairs following the DataHelper style
+  TokenManager.addLocationData = function(dataTable)
+    internal.initDataHelperData()
+    for k, v in pairs(dataTable) do
+      locationData[k] = v
+    end
+  end
+
+  -- Checks to see if the given card has location data in the DataHelper
+  ---@param card Object Card to check for data
+  ---@return Boolean True if this card has data in the helper, false otherwise
+  TokenManager.hasLocationData = function(card)
+    return internal.getLocationData(card) ~= nil
+  end
+
+  -- Copies the data from the DataHelper.  Will only happen once.
```

Comment:
Copying the data was the pattern used by the previous implementation, so I just kept it.

It does simplify access - since the Customs add data to these tables, we'd have to either mutate the base DataHelper (bad) or keep a base list and an added list separately, which would make the lookups harder.